cardano-testnet/src/Testnet/Components/Query.hs

```diff
@@ -233,18 +226,18 @@ findLargestUtxoForPaymentKey epochStateView sbe address =
 -- this number is not attained before two epochs, the test is failed.
 checkDRepsNumber
   :: HasCallStack
-  => MonadCatch m
+  => MonadAssertion m
   => MonadIO m
   => MonadTest m
-  => ShelleyBasedEra ConwayEra -- ^ The era in which the test runs
-  -> NodeConfigFile 'In
-  -> SocketPath
-  -> H.ExecConfig
+  => EpochStateView
+  -> ShelleyBasedEra ConwayEra -- ^ The era in which the test runs
   -> Int
   -> m ()
-checkDRepsNumber sbe configurationFile socketPath execConfig expectedDRepsNb = withFrozenCallStack $
-  checkDRepState sbe configurationFile socketPath execConfig
-    (\m -> if length m == expectedDRepsNb then Just () else Nothing)
+checkDRepsNumber epochStateView sbe expectedDRepsNumber = withFrozenCallStack $
+  checkDRepState epochStateView sbe $ \dreps ->
+    if length dreps == expectedDRepsNumber
+       then Just ()
```

Comment:
What's the point of this? It won't fail unless we hit the termination epoch or `foldEpochState` fails.

Reply:
What do you mean? That's exactly expected. If the number of dreps won't be equal to the expected value within 2 epochs it fails.